File: Arrays/bitwise-operator/singleNumber.c
#include<stdio.h>

int singleNumber(int nums[], int n){
    int ans = 0;
    for(int i = 0; i < n; i++) {
        ans ^= nums[i];
    }
    return ans;
}

int main() {
    int nums[] = {10,1,2,6,2,1,6};
    int n = sizeof(nums) / sizeof(nums[0]);
    int ans = singleNumber(nums, n);
    printf("%d ", ans);//10
}

File: Arrays/bitwise-operator/singleNumber.py

def singleNumber(nums):
    map = {}
    for val in nums:
        map[val] = map.get(val,0) + 1
        
    for num, count in map.items():
        if count == 1:
            return num
    return -1

def singleNumberOptimal(nums):
    ans = 0
    for val in nums:
        ans ^= val
        
    return ans

def main():
    nums = [4,1,2,3,3,1,2]
    ans = singleNumberOptimal(nums)
    ans = singleNumber(nums)
    print(ans)
main()